codebundles/k8s-chaos-flux/runbook.robot

```diff
@@ -0,0 +1,176 @@
+    ...    type=string
+    ...    description=The name of the target resource to run chaos commands in.   
+    ...    pattern=\w*
+    ...    example=deployment/cartservice
+    ...    default=deployment/cartservice
+    ${CHAOS_COMMAND}=    RW.Core.Import User Variable    CHAOS_COMMAND
+    ...    type=string
+    ...    description=The command to run in the target pod.  
+    ...    pattern=\w*
+    ...    example=/bin/sh -c "while true; do yes > /dev/null & done"
+    ...    default=/bin/sh -c "while true; do yes > /dev/null & done"
+    ${CHAOS_COMMAND_LOOP}=    RW.Core.Import User Variable    CHAOS_COMMAND_LOOP
+    ...    type=string
+    ...    description=The number of times to execute this command.  
+    ...    pattern=\d*
+    ...    example=2
+    ...    default=1
+    ${ADDNL_COMMAND}=    RW.Core.Import User Variable    ADDNL_COMMAND
+    ...    type=string
+    ...    description=Run any additional chaos command - verbatim.   
+    ...    pattern=\w*
+    ...    example=kubectl delete svc --all
+    ...    default=kubectl get pods
+    Set Suite Variable    ${kubeconfig}    ${kubeconfig}
+    Set Suite Variable    ${KUBERNETES_DISTRIBUTION_BINARY}    ${KUBERNETES_DISTRIBUTION_BINARY}
+    Set Suite Variable    ${CONTEXT}    ${CONTEXT}
+    Set Suite Variable    ${FLUX_RESOURCE_TYPE}    ${FLUX_RESOURCE_TYPE}
+    Set Suite Variable    ${FLUX_RESOURCE_NAME}    ${FLUX_RESOURCE_NAME}
+    Set Suite Variable    ${FLUX_RESOURCE_NAMESPACE}    ${FLUX_RESOURCE_NAMESPACE}
+    Set Suite Variable    ${TARGET_NAMESPACE}    ${TARGET_NAMESPACE}
+    Set Suite Variable    ${TARGET_RESOURCE}    ${TARGET_RESOURCE}
+    Set Suite Variable    ${CHAOS_COMMAND}    ${CHAOS_COMMAND}
+    Set Suite Variable    ${CHAOS_COMMAND_LOOP}    ${CHAOS_COMMAND_LOOP}
+    Set Suite Variable    ${ADDNL_COMMAND}    ${ADDNL_COMMAND}
+    Set Suite Variable    ${env}    {"KUBECONFIG":"./${kubeconfig.key}"}
+
+*** Tasks ***
+Suspend the Flux Resource Reconciliation
+        [Documentation]    Suspends a flux resource so that it can be manipulated for chaos purposes. 
+        [Tags]    Chaos    Flux    Kubernetes    Resource    Suspend
+        ${suspend_flux_resource}=    RW.CLI.Run Cli
+        ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} patch ${FLUX_RESOURCE_TYPE} ${FLUX_RESOURCE_NAME} -n ${FLUX_RESOURCE_NAMESPACE} --context ${CONTEXT} --type='json' -p='[{"op": "add", "path": "/spec/suspend", "value":true}]'
+        ...    target_service=${kubectl}
+        ...    env=${env}
+        ...    secret_file__kubeconfig=${kubeconfig}
+        ${history}=    RW.CLI.Pop Shell History
+        RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Find Random FluxCD Workload as Chaos Target
+    [Documentation]    Inspects the Flux resource and randomly selects a deployment to tickle. Tehe. Only runs if RANDOMIZE = Yes. 
+    [Tags]    Chaos    Flux    Kubernetes    Resource    Random
+    IF     "${RANDOMIZE}" == "Yes"
```

Comment:
If I opt into running this task, wouldn't I already be affirming that I want the randomized behaviour to run? So is the IF plus randomize config field necessary?